.github/workflows/pr.yaml

```diff
@@ -7,7 +7,7 @@ name: PR workflow running lint checkers, unit and functional tests
 on:
   pull_request:
     types: [ opened, synchronize, reopened ]
-    branches: [ master, main ]
+    branches: [ 3.1 ]
```

Comment:
Based on the conclusion from previous discussion (when we were laying out branching strategy initially), we will continue using mater branch as the default branch and the target for merging pull request. Every time we push new changes to main, we sync the changes to the latest supporting branch (in this case, 3.1) and the release workflow will handle the rest. The philosophy is similar to what's in python-libjuju: develop on the main and merge the changes to version branches).

Reply:
changed the base branch to main. i was not aware of that previous discussion.

Reply:
Thanks. Yes we could've documented this better. I will open a PR afterwards to add the info in README.